compressors/arithmetic_coding.py

```diff
@@ -90,6 +90,70 @@ def update_model(self, s):
                 self.freqs_current.freq_dict[s] = max(f // 2, 1)
 
 
+class AdaptiveOrderKFreqModel(FreqModelBase):
```

Comment:
does it make sense to put the `OrderKFreqModel` etc. in another file, `adaptive_orderk_arithmetic_coding.py`? so that `arithmetic_coding.py` file is cleaner, and we have a focused file dealing with the o`orderk` stuff?

Reply:
I think maybe it makes sense to have the whole `FreqModelBase` class and the inherited classes in another file, maybe in utils? Because frequency model classes can be used with other compressors too. Or in future we can have more complicated probability models. Maybe we should have a `prob_models` file/folder?

Reply:
Left minor comments, looks great overall. Thanks.

Reply:
I've moved to a probability models file as suggested by Pulkit. I kept it in the compressors folder still because the models are a big part of the compressor and deserve to be there.